pkg/controllers/provisioning/scheduling/nodeset.go

```diff
@@ -28,14 +39,29 @@ import (
 )
 
 type NodeSet struct {
```

Comment:
It strikes me that NodeSet is a singleton per execution. Does it make sense to just fold this and the scheduler concept together?

Reply:
I don't think so because the scheduler is re-used.  This also allows a potential defrag to construct a nodeset to evaluate a potential decision.

Reply:
I worry a bit about aspirational abstraction. If we both agreed, I might be more confident about it.

Reply:
I'm fine merging, I mis-remembered  and was thinking Scheduler was a real controller.